### PR TITLE
Fix no module named homeassistant.util.distance

### LINKED ISSUE #18: distance_convert deprecated
Hi, just got my Mila yesterday, installed your integration and all working great! Thanks for your work!

I did notice an entry in the logs and thought I'd log it here...

> Detected integration that uses distance utility. This is deprecated since 2022.10 and will stop working in Home Assistant 2022.4, it should be updated to use unit_conversion.DistanceConverter instead. Please report issue to the custom integration author for mila using this method at custom_components/mila/entities/location/distance_sensor.py, line 34: val = distance_convert(val, LENGTH_KILOMETERS, LENGTH_MILES)

### PR DESCRIPTION
**Pull request by danielnolan**
* homeassistant.util.distance was deprecated in HA 2022.10 and this
  no module error is preventing me from setting up mila on 2023.11.0
* Use DistanceConverter.convert from homeassistant.util.unit_conversion
* Fixes #18

```
2023-11-01 19:17:33.252 ERROR (MainThread) [custom_components.mila.update_coordinator] Unexpected error fetching mila data: No module named 'homeassistant.util.distance'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 290, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/mila/update_coordinator.py", line 105, in _async_update_data
    await self._build_devices(data)
  File "/config/custom_components/mila/update_coordinator.py", line 119, in _build_devices
    self.devices[id] = MilaAppliance(self, self._api, id)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/mila/devices/appliance.py", line 33, in __init__
    super().__init__(coordinator, api, device_id)
  File "/config/custom_components/mila/devices/device.py", line 24, in __init__
    self._build_entities_list()
  File "/config/custom_components/mila/devices/device.py", line 92, in _build_entities_list
    from ..entities import MilaEntity
  File "/config/custom_components/mila/entities/__init__.py", line 3, in <module>
    from .location import *
  File "/config/custom_components/mila/entities/location/__init__.py", line 3, in <module>
    from .distance_sensor import MilaLocationDistanceSensor
  File "/config/custom_components/mila/entities/location/distance_sensor.py", line 9, in <module>
    from homeassistant.util.distance import convert as distance_convert
ModuleNotFoundError: No module named 'homeassistant.util.distance'
```
This is what I was seeing before my change to use the new DistanceConverter
Screenshot 2023-11-01 at 8 23 00 PM

### CODE REVIEW

**Comment by sanghviharshit** [comment]
Thanks for the PR. Do you know if this will break the installs pre 2023.11.0?

**Comment by danielnolan** [comment]
I believe the old module was removed in version 2022.4, so any version between then and now should work with this update.

File: custom_components/mila/entities/location/distance_sensor.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
     LENGTH_MILES
 )
 from homeassistant.util.unit_system import METRIC_SYSTEM
-from homeassistant.util.distance import convert as distance_convert
+from homeassistant.util.unit_conversion import DistanceConversion
```
Comment (kbromberger) [inline custom_components/mila/entities/location/distance_sensor.py]: This still produced an error for me - looks like the static class name in unit_conversion is DistanceConverter. HA Core 2023.11.0

Comment (danielnolan) [reply custom_components/mila/entities/location/distance_sensor.py]: Great catch, I had it correct when I fixed it in my local version, but mistyped it when I cloned the repo and opened the pr.

**Comment by sanghviharshit** [comment]
@kbromberger mind testing this again?

**Comment by kbromberger** [comment]
@sanghviharshit  I can confirm these changes do fix the issue.

**Pull request merged**